### ISSUE TITLE
KNOWN ISSUE: unable to establish ethernet network connection when powered over serial-usb 

### ISSUE THREAD

**Issue by cyberboy666** [cyberboy666/lateral_led_controller#1]
_tl;dr - ethernet will not work when serial programmer is powered unless the serial console is also opened - if not debugging/ updating firmware dont power circuit through the serial port_

## details of issue

internally on the [WT32-ETH01](https://www.aliexpress.com/item/1005007279607510.html) microcontroller board a 50mhz clock signal is passed into pin IO0 to communicate between the ethernet hardware. however this IO0 pin is also used to program the uC (At boot, must be pulled low to program otherwise floating/high to boot normally)

![image](https://github.com/user-attachments/assets/36343715-8f71-4063-b210-6a380958d88b)

the connected [WeMos_CH340G_USB_to_SERIAL](https://www.aliexpress.com/item/1005006642575408.html) handles the IO0 / EN pins with DTR/RTS to enable [automatic bootloading](https://docs.espressif.com/projects/esptool/en/latest/esp32/advanced-topics/boot-mode-selection.html#automatic-bootloader) 

![image](https://github.com/user-attachments/assets/70b56f59-26e0-4d30-8664-94508be068c1)

a correct clock signal (WeMos_CH340G_USB_to_SERIAL unpowered) looks like this (signal between 0.7V and 2.6V):

![wt32-eth01-clock](https://github.com/user-attachments/assets/86343160-2a23-4aca-815d-f87b6ba61bef)

however when the WeMos_CH340G_USB_to_SERIAL (set in 3v3 mode -although same problem when set in 5v mode also) is powered then it pulls the IO0 pin high resulting in a signal like this (signal between 1.3V and 3.9V):

![wt32-eth01-clock3](https://github.com/user-attachments/assets/b7f2d95d-a9ca-491f-bd82-587d2831551a)

this does not allow the ethernet connection to be made...

## how to avoid this problem

the easy work-around for this is when using this circuit to only power it through the barrel jack POWER IN - not via the _serial programmer_ . this is recommended anyway.

you should only use the _serial programmer_ when uploading new code or when debugging something.

interestingly it is also not a problem when monitoring the serial console print out for debugging since opening the serial monitor triggers the WeMos_CH340G_USB_to_SERIAL to pull IO0 pin low which allows the ethernet serial to connect
